crates/ra_ide_db/src/search.rs

```diff
@@ -237,9 +239,8 @@ impl Definition {
                 // FIXME: reuse sb
                 // See https://github.com/rust-lang/rust/pull/68198#issuecomment-574269098
 
-                if let Some(d) = classify_name_ref(&sema, &name_ref) {
-                    let d = d.definition();
-                    if &d == self {
+                match classify_name_ref(&sema, &name_ref) {
+                    Some(NameRefClass::Definition(def)) if &def == self => {
```

Comment:
I just noticed I'm getting some bad renames running the full binary with this patch when I rename the `Kind` field on the `Reference` struct. I don't have time to dig in now but my first guess is the `&def == self` check is pulling more weight than I realized and I need something similar on the other branch of the match.

Reply:
Yeah, in `NameRefClass::FieldShorthand` you have two different cases: either you are renaming the field, or the local, and they need to be handled differently.

Reply:
I pushed a fix and test for the bad behavior. Everything seems to be working now. 

I'm not 100% clear on the `access` field. In the new cases should it always be `Some(ReferenceAccess::Read)` instead of calculated from `reference_access`?

Reply:
i guess it should be write for ref to a field and read for ref to a local, but reference accesses only affect UI presentation, so that's not that important at the moment.